gem installation fails due to `-Werror=calloc-transposed-args`

Hello!

Building the `prism` gem at v0.24.0 fails on Fedora rawhide, where I have `cc (GCC) 14.0.1 20240316 (Red Hat 14.0.1-0)` —

```shell-session
$ gem install prism
Building native extensions. This could take a while...
ERROR:  Error installing prism:
	ERROR: Failed to build gem native extension.

    current directory: /home/jashank/.gem/ruby/3.3.0/gems/prism-0.24.0/ext/prism
/usr/bin/ruby -I/usr/share/rubygems extconf.rb
checking for prism.h in /home/jashank/.gem/ruby/3.3.0/gems/prism-0.24.0/include... yes
checking for prism/extension.h in /home/jashank/.gem/ruby/3.3.0/gems/prism-0.24.0/ext... yes
checking for whether -fvisibility=hidden is accepted as CFLAGS... yes
compiling build/static/regexp.o
compiling build/static/util/pm_char.o
compiling build/static/util/pm_list.o
compiling build/static/util/pm_strncasecmp.o
compiling build/static/util/pm_memchr.o
compiling build/static/util/pm_buffer.o
compiling build/static/util/pm_string_list.o
compiling build/static/util/pm_newline_list.o
compiling build/static/util/pm_state_stack.o
compiling build/static/util/pm_strpbrk.o
compiling build/static/util/pm_constant_pool.o
compiling build/static/util/pm_string.o
compiling build/static/diagnostic.o
src/diagnostic.c: In function ‘pm_diagnostic_list_append’:
src/diagnostic.c:328:69: error: ‘calloc’ sizes specified with ‘sizeof’ in the earlier argument and not in the later argument [-Werror=calloc-transposed-args]
  328 |     pm_diagnostic_t *diagnostic = (pm_diagnostic_t *) calloc(sizeof(pm_diagnostic_t), 1);
      |                                                                     ^~~~~~~~~~~~~~~
src/diagnostic.c:328:69: note: earlier argument should specify number of elements, later size of each element
src/diagnostic.c: In function ‘pm_diagnostic_list_append_format’:
src/diagnostic.c:359:69: error: ‘calloc’ sizes specified with ‘sizeof’ in the earlier argument and not in the later argument [-Werror=calloc-transposed-args]
  359 |     pm_diagnostic_t *diagnostic = (pm_diagnostic_t *) calloc(sizeof(pm_diagnostic_t), 1);
      |                                                                     ^~~~~~~~~~~~~~~
src/diagnostic.c:359:69: note: earlier argument should specify number of elements, later size of each element
cc1: all warnings being treated as errors
make: *** [Makefile:54: build/static/diagnostic.o] Error 1
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
	--with-opt-dir
	--without-opt-dir
	--with-opt-include=${opt-dir}/include
	--without-opt-include
	--with-opt-lib=${opt-dir}/lib64
	--without-opt-lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/usr/bin/$(RUBY_BASE_NAME)
	--enable-debug-mode-build
	--disable-debug-mode-build
extconf.rb:41:in `system': Command failed with exit 2: make (RuntimeError)
	from extconf.rb:41:in `block in make'
	from extconf.rb:40:in `chdir'
	from extconf.rb:40:in `make'
	from extconf.rb:92:in `<main>'

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  /home/jashank/.gem/ruby/3.3.0/extensions/x86_64-linux/3.3.0/prism-0.24.0/mkmf.log

extconf failed, exit code 1

Gem files will remain installed in /home/jashank/.gem/ruby/3.3.0/gems/prism-0.24.0 for inspection.
Results logged to /home/jashank/.gem/ruby/3.3.0/extensions/x86_64-linux/3.3.0/prism-0.24.0/gem_make.out
```

A similar issue also exists in `src/prism.c`, if I attempt to build from the repository source (at 02d93793da61ca099694a5dd427bbffa01d29d24):

```
compiling build/static/prism.o
src/prism.c: In function ‘parser_lex_magic_comment’:
src/prism.c:6776:67: error: ‘calloc’ sizes specified with ‘sizeof’ in the earlier argument and not in the later argument [-Werror=calloc-transposed-args]
 6776 |         if ((magic_comment = (pm_magic_comment_t *) calloc(sizeof(pm_magic_comment_t), 1)) != NULL) {
      |                                                                   ^~~~~~~~~~~~~~~~~~
src/prism.c:6776:67: note: earlier argument should specify number of elements, later size of each element
src/prism.c: In function ‘parser_comment’:
src/prism.c:8208:60: error: ‘calloc’ sizes specified with ‘sizeof’ in the earlier argument and not in the later argument [-Werror=calloc-transposed-args]
 8208 |     pm_comment_t *comment = (pm_comment_t *) calloc(sizeof(pm_comment_t), 1);
      |                                                            ^~~~~~~~~~~~
src/prism.c:8208:60: note: earlier argument should specify number of elements, later size of each element
cc1: all warnings being treated as errors
make: *** [Makefile:54: build/static/prism.o] Error 1
```

--- a/src/prism.c
+++ b/src/prism.c
@@ -7582,7 +7582,7 @@ parser_lex_magic_comment(pm_parser_t *parser, bool semantic_token_seen) {
 
         // Allocate a new magic comment node to append to the parser's list.
         pm_magic_comment_t *magic_comment;
-        if ((magic_comment = (pm_magic_comment_t *) xcalloc(sizeof(pm_magic_comment_t), 1)) != NULL) {
+        if ((magic_comment = (pm_magic_comment_t *) xcalloc(1, sizeof(pm_magic_comment_t))) != NULL) {
             magic_comment->key_start = key_start;
             magic_comment->value_start = value_start;
             magic_comment->key_length = (uint32_t) key_length;
@@ -9077,7 +9077,7 @@ parser_lex_callback(pm_parser_t *parser) {
  */
 static inline pm_comment_t *
 parser_comment(pm_parser_t *parser, pm_comment_type_t type) {
-    pm_comment_t *comment = (pm_comment_t *) xcalloc(sizeof(pm_comment_t), 1);
+    pm_comment_t *comment = (pm_comment_t *) xcalloc(1, sizeof(pm_comment_t));
     if (comment == NULL) return NULL;
 
     *comment = (pm_comment_t) {

--- a/templates/src/diagnostic.c.erb
+++ b/templates/src/diagnostic.c.erb
@@ -383,7 +383,7 @@ pm_diagnostic_level(pm_diagnostic_id_t diag_id) {
  */
 bool
 pm_diagnostic_list_append(pm_list_t *list, const uint8_t *start, const uint8_t *end, pm_diagnostic_id_t diag_id) {
-    pm_diagnostic_t *diagnostic = (pm_diagnostic_t *) xcalloc(sizeof(pm_diagnostic_t), 1);
+    pm_diagnostic_t *diagnostic = (pm_diagnostic_t *) xcalloc(1, sizeof(pm_diagnostic_t));
     if (diagnostic == NULL) return false;
 
     *diagnostic = (pm_diagnostic_t) {
@@ -415,7 +415,7 @@ pm_diagnostic_list_append_format(pm_list_t *list, const uint8_t *start, const ui
         return false;
     }
 
-    pm_diagnostic_t *diagnostic = (pm_diagnostic_t *) xcalloc(sizeof(pm_diagnostic_t), 1);
+    pm_diagnostic_t *diagnostic = (pm_diagnostic_t *) xcalloc(1, sizeof(pm_diagnostic_t));
     if (diagnostic == NULL) {
         return false;
     }

